cmd/sonobuoy/app/args.go

```diff
@@ -301,11 +301,12 @@ func AddRBACModeFlags(mode *RBACMode, flags *pflag.FlagSet, defaultMode RBACMode
 }
 
 // AddSkipPreflightFlag adds a boolean flag to skip preflight checks.
-func AddSkipPreflightFlag(flag *bool, flags *pflag.FlagSet) {
-	flags.BoolVar(
-		flag, "skip-preflight", false,
+func AddSkipPreflightFlag(flag *[]string, flags *pflag.FlagSet) {
+	flags.StringSliceVar(
+		flag, "skip-preflight", []string{""},
```

Comment:
This is currently the cause of the test failure. The default value here is a single value (the empty string).

What you're meaning to do here is use `[]string{}` (a slice of strings with no values at all).

Otherwise what was happening was that you're iterating over this list later and seeing the empty string given and erroring out.